src/store/FilingHistoryList/actions.ts

```diff
@@ -0,0 +1,196 @@
+import { FilingSubTypes } from '@/enums'
+import { FilingTypes } from '@bcrs-shared-components/enums'
+import { ApiFilingIF, CommentIF, DocumentIF } from '@/interfaces'
+import { DateUtilities, EnumUtilities, LegalServices } from '@/services/'
+
+export default {
+  /**
+   * Fetches the filings list from the Legal API and, if successful, triggers some mutations.
+   * @param context the Vuex context (passed in automatically)
+   */
+  loadFilings ({ dispatch }): Promise<any> {
+    // need to return a promise because action is called via dispatch
+    return new Promise((resolve, reject) => {
+      const businessId = sessionStorage.getItem('BUSINESS_ID')
+      const tempRegNumber = sessionStorage.getItem('TEMP_REG_NUMBER')
+
+      // if there is no business id, return error
+      if (!businessId && !tempRegNumber) {
+        reject(new Error('Missing business id or temp reg number'))
+        return
+      }
+
+      LegalServices.fetchFilings(businessId || tempRegNumber)
```

Comment:
Would it be possible to pass the `businessId` or `tempRegNumber` to the action as a parameter?  Then we wouldn't have to check for its existence, and it makes `loadFilings()` a pure function.

Reply:
... well relatively pure for a function that gets from a database. :-)

Reply:
I think it's possible to pass in 1 extra parameter, but it could be an object with 2 properties.

I don't think businessId or tempRegNumber need to be parameterized, since they are set once when the app starts and don't change thereafter.

I would love to get these values from somewhere else than session state every time. Maybe the store, if they're set before the getter sees them.

However, the focus of this ticket was the ledger refactoring and I consider this a bit out of scope. Yes, I'll fix small things along the way, but I tried to leave existing functions alone as much as possible. In the example above, I pretty copied the code out of App.vue.

We'll see how today goes. Maybe I'll have time to update this and maybe higher priority things will take precedence.

Reply:
> I don't think businessId or tempRegNumber need to be parameterized, since they are set once when the app starts and don't change thereafter.

Didn't we groom bcgov/entity#15657 to pass businessId or the tempRegNumber as parameters from VueRouter?  I thought we agreed, it would be better to follow best practices than a roll-your-own approach?

Reply:
Yes, we groomed that ticket for Edit UI, and I agree that we should look at that approach.

Reply:
This action method is now passed "id" instead of getting it from session storage. Good to resolve this conversation?